Tasks/Common/azure-arm-rest-v2/azure-arm-common.ts

```diff
@@ -83,10 +84,18 @@ export class ApplicationTokenCredentials {
         }
         
         this.isADFSEnabled = isADFSEnabled;
+        this.accessToken = access_token;
 
     }
 
     public getToken(force?: boolean): Q.Promise<string> {
+        if (!!this.accessToken && !force) {
+            tl.debug("==================== USING ENDPOINT PROVIDED ACCESS TOKEN ====================");
```

Comment:
Suggestion: make log more verbose, indicating that token was picked from key "apiToken"